network-api/networkapi/wagtailpages/pagemodels/buyersguide/products.py

```diff
@@ -960,6 +968,12 @@ def get_primary_related_articles(self) -> models.QuerySet['BuyersGuideArticlePag
     def get_secondary_related_articles(self) -> models.QuerySet['BuyersGuideArticlePage']:
         return self.get_related_articles()[3:]
 
+    def check_for_categories_featuring_cta(self):
+        if self.product_categories.filter(category__show_cta=True):
```

Comment:
**Question** Have you considered the use of [`.exists()`](https://docs.djangoproject.com/en/4.1/ref/models/querysets/#exists)?

Reply:
Great read Tibor, thanks for the link! I have added this as it seems like a good idea.